Application/Api/Billing/Bookkeeping/ApiBasketRepaymentAddPerson.php

```diff
@@ -0,0 +1,903 @@
+    {
+
+        return (new BlockReceiver($Content))->setIdentifier('Block'.$Identifier);
+    }
+
+    /**
+     * @param string $Content
+     *
+     * @return BlockReceiver
+     */
+    public static function receiverTableLayout($Content = '')
+    {
+
+        return (new BlockReceiver($Content))->setIdentifier('BasketVerificationLayout');
+    }
+
+    /**
+     * @return InlineReceiver
+     */
+    public static function receiverService()
+    {
+
+        return (new InlineReceiver())->setIdentifier('Service');
+    }
+
+    /**
+     * @param string $Content
+     * @param string $Identifier
+     *
+     * @return InlineReceiver
+     */
+    public static function receiverItemPrice($Content = '', $Identifier = '')
+    {
+
+        return (new InlineReceiver($Content))->setIdentifier('Price'.$Identifier);
+    }
+
+    /**
+     * @param string $BasketId
+     * @param string $Identifier
+     *
+     * @return Pipeline
+     */
+    public static function pipelineReloadTable($BasketId = '', $Identifier = '')
+    {
+        $Pipeline = new Pipeline(false);
+        // reload columns
+        // reload Summary column
+        $Emitter = new ServerEmitter(self::receiverTableLayout(), self::getEndpoint());
+        $Emitter->setGetPayload(array(
+            self::API_TARGET => 'getTableLayout'
+        ));
+        $Emitter->setPostPayload(array(
+            'BasketId' => $BasketId
+        ));
+        $Pipeline->appendEmitter($Emitter);
+        if($Identifier){
+            $Pipeline->appendEmitter((new CloseModal(self::receiverModal('', $Identifier)))->getEmitter());
+        }
+        return $Pipeline;
+    }
+
+    /**
+     * @param string $BasketId
+     *
+     * @return Layout|string
+     */
+    public function getTableLayout($BasketId = '')
+    {
+
+        if($BasketId){
+            return Basket::useFrontend()->getBasketVerificationRepaymentLayout($BasketId);
+        }
+        return 'Fehler';
+    }
+
+    /**
+     * @param string $Identifier
+     * @param string $BasketId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineOpenAddPersonModal($Identifier = '', $BasketId = '')
+    {
+
+        $Receiver = self::receiverModal(null, $Identifier);
+        $Pipeline = new Pipeline();
+        $Emitter = new ServerEmitter($Receiver, self::getEndpoint());
+        $Emitter->setGetPayload(array(
+            self::API_TARGET => 'showSearchModal'
+        ));
+        $Emitter->setPostPayload(array(
+            'Identifier' => $Identifier,
+            'BasketId'   => $BasketId,
+        ));
+        $Pipeline->appendEmitter($Emitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param string $Identifier
+     * @param string $searchIdentifier
+     * @param string $BasketId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineSearch($Identifier, $searchIdentifier = '', $BasketId = '')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [pipelineSearch accesses the super-global variable $_POST.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)